data_structures/linked_list/reverse_linked_list.py

```diff
@@ -0,0 +1,49 @@
+class Node:
+    def __init__(self, data):
+        self.data = data
+        self.next = None
+
+
+class LinkedList:
+    def __init__(self):
+        self.head = None
+
+    def reverse(self, head):
+
+        if head is None or head.next is None:
+            return head
+
+        rest = self.reverse(head.next)
+
+        head.next.next = head
+        head.next = None
+
+        return rest
+
+    def __str__(self):
+        linkedListStr = ""
+        temp = self.head
+        while temp:
+            linkedListStr = linkedListStr + str(temp.data) + " "
+            temp = temp.next
+        return linkedListStr
+
+    def push(self, data):
+        temp = Node(data)
+        temp.next = self.head
+        self.head = temp
+
+
+linkedList = LinkedList()
+linkedList.push(20)
+linkedList.push(4)
+linkedList.push(15)
+linkedList.push(85)
+
+print("Given linked list")
+print(linkedList)
+
+linkedList.head = linkedList.reverse(linkedList.head)
+
+print("Reversed linked list")
+print(linkedList)
```

Comment:
Please read `CONTRIBUTING.md` to see why all these changes are required...
```suggestion
if __name__ == "__main__":
    linked_list = LinkedList()
    linked_list.push(20)
    linked_list.push(4)
    linked_list.push(15)
    linked_list.push(85)

    print("Given linked list")
    print(linked_list)

    linked_list.head = linked_list.reverse(linked_list.head)

    print("Reversed linked list")
    print(linked_list)
```